lib/ResolverFactory.js

```diff
@@ -61,7 +61,9 @@ const convertToResolveOptions = resolveOptionsWithDepType => {
 		);
 
 	return removeOperations(
-		resolveByProperty(options, "byDependency", dependencyType)
+		resolveByProperty(options, "byDependency", dependencyType),
+		// Keep the `unsafeCache` because it can be a `Proxy`
+		["unsafeCache"]
```

Comment:
We use `removeOperations` only for resolver options and there is only one option which can be `object` - `unsafeCache`, also we don't need to remove `...`/undefined/etc from it, so, let's keep this object as is